diff -urN oai-cn5g-upf.orig/build/scripts/build_helper.upf oai-cn5g-upf/build/scripts/build_helper.upf
--- oai-cn5g-upf.orig/build/scripts/build_helper.upf	2026-05-31 13:20:37.000000000 +0900
+++ oai-cn5g-upf/build/scripts/build_helper.upf	2026-05-31 14:13:08.573036204 +0900
@@ -275,6 +275,9 @@
       "ubuntu20.04")
         specific_packages="libconfig++-dev libasio-dev libboost-all-dev libasan5"
         ;;
+      "ubuntu24.04")
+        specific_packages="libconfig++-dev libasio-dev libboost-all-dev libasan8"
+        ;;
       *)
         specific_packages="libconfig++-dev libasio-dev libboost-all-dev libasan?"
         ;;
@@ -393,6 +396,12 @@
         install_libyaml_cpp_from_git $1 $2
         ret=$?;[[ $ret -ne 0 ]] && return $ret
         ;;
+      "ubuntu24.04")
+        # probably 0.8.0
+        #$SUDO $INSTALLER install $OPTION libyaml-cpp0.8 libyaml-cpp-dev
+        install_libyaml_cpp_from_git $1 $2
+        ret=$?;[[ $ret -ne 0 ]] && return $ret
+        ;;
     esac
   elif [[ "$OS_BASEDISTRO" == "fedora" ]]; then
     # definitely, 0.6.2 as PPA installed is not enough
@@ -498,7 +507,7 @@
 
 check_platform_for_lttng() {
   local distribution=$(get_distribution_release)
-  if [[ "$distribution" == "ubuntu18.04" || "$distribution" == "ubuntu20.04" || "$distribution" == "ubuntu22.04" ]]; then
+  if [[ "$distribution" == "ubuntu18.04" || "$distribution" == "ubuntu20.04" || "$distribution" == "ubuntu22.04" || "$distribution" == "ubuntu24.04" ]]; then
       return 1
   fi
   return 0
diff -urN oai-cn5g-upf.orig/docker/Dockerfile.upf.ubuntu oai-cn5g-upf/docker/Dockerfile.upf.ubuntu
--- oai-cn5g-upf.orig/docker/Dockerfile.upf.ubuntu	2026-08-11 11:31:56.074826110 +0900
+++ oai-cn5g-upf/docker/Dockerfile.upf.ubuntu	2026-08-11 11:35:29.279719217 +0900
@@ -13,7 +13,7 @@
 #---------------------------------------------------------------------
 # BASE IMAGE
 #---------------------------------------------------------------------
-ARG BASE_IMAGE=ubuntu:jammy
+ARG BASE_IMAGE=ubuntu:noble
 FROM $BASE_IMAGE AS oai-upf-base
 
 ENV DEBIAN_FRONTEND=noninteractive
@@ -50,6 +50,23 @@
 FROM oai-upf-base AS oai-upf-builder
 # Copy the rest of source code
 COPY . /openair-upf
+
+
+# Remove the deprecated NPN argument to add support for Ubuntu 24.04
+RUN sed -i \
+  's/cpr::ssl::ALPN{false}, cpr::ssl::NPN{false}/cpr::ssl::ALPN{false}/g' \
+  /openair-upf/src/common-src/http/http_client.cpp
+
+# Ubuntu 24.04 - <cstdint> is no longer implicitly included by standard headers
+RUN sed -i '1i #include <cstdint>' \
+  /openair-upf/src/common-src/utils/fqdn.hpp
+
+RUN sed -i '1i #include <cstdint>' \
+  /openair-upf/src/common-src/utils/output_wrapper.hpp
+
+RUN sed -i '1i #include <cstdint>' \
+  /openair-upf/src/common-src/utils/mime_parser.hpp
+
 # Building UPF
 WORKDIR /openair-upf/build/scripts
 ARG BUILD_ARGS=""
@@ -95,12 +112,14 @@
     netbase \
     # Ubuntu 20 --> libasan5
     # Ubuntu 22 --> libasan6
+    # Ubuntu 24 --> libasan8
     #      libasan? \
     libgssapi-krb5-2 \
     # Ubuntu 20 --> libldap-2.4-2
     # Ubuntu 22 --> libldap-2.5-0
+    # Ubuntu 24 --> libldap-2.6-7
     libldap-2.?-? \
-    libgoogle-glog0v5 \
+    libgoogle-glog0v6 \
     libdouble-conversion3 \
     libconfig++9v5 \
     librtmp1 \
@@ -109,7 +128,8 @@
     libcurl?-gnutls \
     # Ubuntu 20 --> boost71
     # Ubuntu 22 --> boost74
-    libboost-thread1.7?.0 \
+    # Ubuntu 24 --> boost83
+    libboost-thread1.83.0 \
     # lttng
     && if [ "$ENABLE_LTTNG" = "true" ]; then DEBIAN_FRONTEND=noninteractive apt-get install --yes liblttng-ust-dev; fi \
     && rm -rf /var/lib/apt/lists/*
@@ -128,11 +148,11 @@
 COPY --from=oai-upf-builder \
     /usr/local/lib/libfmt.so \
     /usr/local/lib/libyaml-cpp.so.0.? \
-    /usr/local/lib/libcpr.so.1 \    
+    /usr/local/lib/libcpr.so.1* \    
     # Ubuntu 20 --> libbfd-2.34
     # Ubuntu 22 --> libbfd-2.38
-    /usr/lib/?-linux-gnu/libbfd-2.3?-system.so \
-    /usr/lib/?-linux-gnu/libopcodes-2.3?-system.so \
+    /usr/lib/?-linux-gnu/libbfd-2.*?-system.so \
+    /usr/lib/?-linux-gnu/libopcodes-2.*?-system.so \
     /openair-upf/build/ext/libbpf/src/libbpf.so.1 \
     # /openair-upf/build/upf/build/upf_app/utils/libupfutils.so \
     ./
