diff -urN common-src.orig/3gpp/3gpp_29.244.h common-src/3gpp/3gpp_29.244.h
--- common-src.orig/3gpp/3gpp_29.244.h	2026-01-13 23:31:08.000000000 +0900
+++ common-src/3gpp/3gpp_29.244.h	2026-01-13 23:36:27.000000000 +0900
@@ -1893,7 +1893,9 @@
   _3GPP_INTERFACE_TYPE_N3_TRUSTED_NON_3GPP_ACCESS   = 12,
   _3GPP_INTERFACE_TYPE_N3_UNTRUSTED_NON_3GPP_ACCESS = 13,
   _3GPP_INTERFACE_TYPE_N3                           = 14,
-  _3GPP_INTERFACE_TYPE_N9                           = 15
+  _3GPP_INTERFACE_TYPE_N9                           = 15,
+  _3GPP_INTERFACE_TYPE_SGI                          = 16,
+  _3GPP_INTERFACE_TYPE_N6                           = 17
 };
 
 typedef struct _3gpp_interface_type_s {
diff -urN common-src.orig/pfcp/msg_pfcp.hpp common-src/pfcp/msg_pfcp.hpp
--- common-src.orig/pfcp/msg_pfcp.hpp	2026-01-13 23:31:09.000000000 +0900
+++ common-src/pfcp/msg_pfcp.hpp	2026-01-14 20:03:26.000000000 +0900
@@ -1707,7 +1707,8 @@
         qfi(),
         framed_route(),
         framed_routing(),
-        framed_ipv6_route() {}
+        framed_ipv6_route(),
+        _3gpp_interface_type() {}
 
   pdi(const pdi& p)
       : source_interface(p.source_interface),
@@ -1722,7 +1723,8 @@
         qfi(p.qfi),
         framed_route(p.framed_route),
         framed_routing(p.framed_routing),
-        framed_ipv6_route(p.framed_ipv6_route) {}
+        framed_ipv6_route(p.framed_ipv6_route),
+        _3gpp_interface_type(p._3gpp_interface_type) {}
 
   // virtual ~pdi() {};
   void set(const pfcp::source_interface_t& v) {
@@ -2019,6 +2021,7 @@
   std::pair<bool, pfcp::header_enrichment_t> header_enrichment;
   std::pair<bool, pfcp::pfcpsmreq_flags_t> pfcpsmreq_flags;
   std::pair<bool, pfcp::traffic_endpoint_id_t> linked_traffic_endpoint_id;
+  std::pair<bool, pfcp::_3gpp_interface_type_t> _3gpp_interface_type;
 
   update_forwarding_parameters()
       : destination_interface(),
@@ -2028,7 +2031,8 @@
         forwarding_policy(),
         header_enrichment(),
         pfcpsmreq_flags(),
-        linked_traffic_endpoint_id() {}
+        linked_traffic_endpoint_id(),
+        _3gpp_interface_type() {}
 
   update_forwarding_parameters(const update_forwarding_parameters& u)
       : destination_interface(u.destination_interface),
@@ -2038,7 +2042,8 @@
         forwarding_policy(u.forwarding_policy),
         header_enrichment(u.header_enrichment),
         pfcpsmreq_flags(u.pfcpsmreq_flags),
-        linked_traffic_endpoint_id(u.linked_traffic_endpoint_id) {}
+        linked_traffic_endpoint_id(u.linked_traffic_endpoint_id),
+        _3gpp_interface_type(u._3gpp_interface_type) {}
 
   // virtual ~update_forwarding_parameters() {};
   void set(const pfcp::destination_interface_t& v) {
@@ -2073,6 +2078,10 @@
     linked_traffic_endpoint_id.first  = true;
     linked_traffic_endpoint_id.second = v;
   }
+  void set(const pfcp::_3gpp_interface_type_t& v) {
+    _3gpp_interface_type.first  = true;
+    _3gpp_interface_type.second = v;
+  }
 
   bool get(pfcp::destination_interface_t& v) const {
     if (destination_interface.first) {
@@ -2130,6 +2139,13 @@
     }
     return false;
   }
+  bool get(pfcp::_3gpp_interface_type_t& v) const {
+    if (_3gpp_interface_type.first) {
+      v = _3gpp_interface_type.second;
+      return true;
+    }
+    return false;
+  }
 };
 
 //------------------------------------------------------------------------------
@@ -2145,6 +2161,7 @@
   std::pair<bool, pfcp::header_enrichment_t> header_enrichment;
   std::pair<bool, pfcp::traffic_endpoint_id_t> linked_traffic_endpoint_id;
   std::pair<bool, pfcp::proxying_t> proxying;
+  std::pair<bool, pfcp::_3gpp_interface_type_t> _3gpp_interface_type;
 
   forwarding_parameters()
       : destination_interface(),
@@ -2155,7 +2172,8 @@
         forwarding_policy(),
         header_enrichment(),
         linked_traffic_endpoint_id(),
-        proxying() {}
+        proxying(),
+        _3gpp_interface_type() {}
 
   forwarding_parameters(const forwarding_parameters& f)
       : destination_interface(f.destination_interface),
@@ -2166,7 +2184,8 @@
         forwarding_policy(f.forwarding_policy),
         header_enrichment(f.header_enrichment),
         linked_traffic_endpoint_id(f.linked_traffic_endpoint_id),
-        proxying(f.proxying) {}
+        proxying(f.proxying),
+        _3gpp_interface_type(f._3gpp_interface_type) {}
 
   bool update(const update_forwarding_parameters& u) {
     if (u.destination_interface.first) set(u.destination_interface.second);
@@ -2217,6 +2236,10 @@
     proxying.first  = true;
     proxying.second = v;
   }
+  void set(const pfcp::_3gpp_interface_type_t& v) {
+    _3gpp_interface_type.first  = true;
+    _3gpp_interface_type.second = v;
+  }
 
   bool get(pfcp::destination_interface_t& v) const {
     if (destination_interface.first) {
@@ -2280,6 +2303,13 @@
       return true;
     }
     return false;
+  }
+  bool get(pfcp::_3gpp_interface_type_t& v) const {
+    if (_3gpp_interface_type.first) {
+      v = _3gpp_interface_type.second;
+      return true;
+    }
+    return false;
   }
 };
 
