diff -urN oai-cn5g-upf.orig/src/udp/udp.cpp oai-cn5g-upf/src/udp/udp.cpp
--- oai-cn5g-upf.orig/src/udp/udp.cpp	2026-06-06 02:44:30.000000000 +0900
+++ oai-cn5g-upf/src/udp/udp.cpp	2026-06-06 08:56:25.903145134 +0900
@@ -113,7 +113,8 @@
              &worker->r_endpoint.addr_storage_len)) > 0) {
       work_pool_->write(worker);
     } else {
-      Logger::udp().error("Recvfrom failed %s", strerror(errno));
+      if (errno != 0)
+        Logger::udp().error("Recvfrom failed %s", strerror(errno));
       free_pool_->write(worker);
     }
     if (terminateRL_) {
diff -urN oai-cn5g-upf.orig/src/upf_app/kernel/xdp/upf_xdp_kern.c oai-cn5g-upf/src/upf_app/kernel/xdp/upf_xdp_kern.c
--- oai-cn5g-upf.orig/src/upf_app/kernel/xdp/upf_xdp_kern.c	2026-06-06 02:44:30.000000000 +0900
+++ oai-cn5g-upf/src/upf_app/kernel/xdp/upf_xdp_kern.c	2026-06-07 05:30:38.792459577 +0900
@@ -81,7 +81,6 @@
 /* Maps and definitions */
 #include "upf_xdp_maps.h"
 #include "interfaces.h"
-#include "pfcp_session_eth_pdu.h"
 #include "sdf_filter.h"
 #include "framed_routing_bpf.h"
 #include "xdp_stats_kern.h"
@@ -1574,7 +1573,6 @@
 SEC("xdp")
 int xdp_uplink(struct xdp_md* ctx) {
   bpf_debug("========< XDP Uplink: N3 --> N6 >========");
-  int action = XDP_PASS;
 
   /*
     |-----------------------------------------------------------------------|
@@ -1611,11 +1609,8 @@
         "No "
         "session for UE IP: %pI4",
         ue_ip);
-    action = xdp_stats_record_action(ctx, XDP_PASS);
-    if (action != XDP_PASS) {
-      return action;
-    }
-    return entry_point_uplink__eth_pdu(ctx);
+    return xdp_stats_record_action(ctx, XDP_PASS);
+    // return XDP_PASS;
   }
 
   u64 seid                            = session->seid;
@@ -1672,11 +1667,7 @@
     case RET_FAILURE:
     default: {
       bpf_debug("PASS: something went wrong! pass packet to kernel");
-      action = xdp_stats_record_action(ctx, XDP_PASS);
-      if (action != XDP_PASS) {
-        return action;
-      }
-      return entry_point_uplink__eth_pdu(ctx);
+      return xdp_stats_record_action(ctx, XDP_PASS);
     }
   }
 }
@@ -1835,7 +1826,6 @@
 SEC("xdp")
 int xdp_downlink(struct xdp_md* ctx) {
   bpf_debug("========< XDP Downlink: N6 --> N3 >========");
-  int action = XDP_PASS;
 
   /*
    |-----------------------------------------------------------------------|
@@ -1865,11 +1855,7 @@
   if (!session) {
     bpf_debug(
         "PFCP Session Lookup (Find PFCP session with matching PDRs) failed");
-    action = xdp_stats_record_action(ctx, XDP_PASS);
-    if (action != XDP_PASS) {
-      return action;
-    }
-    return entry_point_downlink__eth_pdu(ctx);
+    return xdp_stats_record_action(ctx, XDP_PASS);
   }
 
   u64 seid                            = session->seid;
@@ -1924,11 +1910,7 @@
     }
     default: {
       bpf_debug("PASS: something went wrong! pass packet to kernel");
-      action = xdp_stats_record_action(ctx, XDP_PASS);
-      if (action != XDP_PASS) {
-        return action;
-      }
-      return entry_point_downlink__eth_pdu(ctx);
+      return xdp_stats_record_action(ctx, XDP_PASS);
     }
   }
 }
